platform-sdk/platform-apps/tests/ConsistencyTestingTool/src/main/java/com/swirlds/demo/consistency/ConsistencyTestingToolState.java

```diff
@@ -82,26 +81,64 @@ private static class ClassVersion {
     /**
      * Constructor
      */
-    public ConsistencyTestingToolState() {
+    public ConsistencyTestingToolState(@NonNull final VirtualMap virtualMap) {
+        super(virtualMap);
         transactionHandlingHistory = new TransactionHandlingHistory();
         transactionsAwaitingPostHandle = ConcurrentHashMap.newKeySet();
         logger.info(STARTUP.getMarker(), "New State Constructed.");
     }
 
+    /**
+     * Copy constructor
+     *
+     * @param that the state to copy
+     */
+    private ConsistencyTestingToolState(@NonNull final ConsistencyTestingToolState that) {
+        super(Objects.requireNonNull(that));
+        this.stateLong = that.stateLong;
+        this.roundsHandled = that.roundsHandled;
+        this.transactionHandlingHistory = that.transactionHandlingHistory;
+        this.transactionsAwaitingPostHandle = that.transactionsAwaitingPostHandle;
+    }
+
+    @Override
+    protected ConsistencyTestingToolState copyingConstructor() {
+        return new ConsistencyTestingToolState(this);
+    }
+
+    @Override
+    protected ConsistencyTestingToolState newInstance(@NonNull VirtualMap virtualMap) {
+        return new ConsistencyTestingToolState(virtualMap);
+    }
+
     /**
      * Initialize the state
      */
     void initState(Path logFilePath) {
-        final StringLeaf stateLongLeaf = getChild(STATE_LONG_INDEX);
-        if (stateLongLeaf != null && stateLongLeaf.getLabel() != null) {
-            this.stateLong = Long.parseLong(stateLongLeaf.getLabel());
+        final var schema = new V0670ConsistencyTestingToolSchema();
+        schema.statesToCreate().stream()
+                .sorted(Comparator.comparing(StateDefinition::stateId))
```

Comment:
Why does it have to be sorted?

Reply:
I think in this case, it doesn't matter. I did it because it looks like that it is a basic rule of thumb when we initialize the state. That is because when we initialize the state, we can call migrate method on schema, and sometimes, that migrate method would reference a schema, which already should've been initialized (I just stumbled upon such case not long ago).